### ISSUE TITLE
Collapsing every sub-level of the mindmap with a command

### ISSUE THREAD

**Issue by santiyounger** [lynchjames/obsidian-mind-map#31]
Hey, thank you so much for the excellent plug-in
Would it be possible to have a command (hotkey) to collapse / fold everything in a mindmap

![image](https://user-images.githubusercontent.com/51902730/107812335-8400a700-6d3d-11eb-82cb-53728d6d63e7.png)

to this

![image](https://user-images.githubusercontent.com/51902730/107812382-95e24a00-6d3d-11eb-8f60-5dd07cc82b23.png)
  
 However of course that can normally be achieved by clicking on the top level circles (in this case "1" and "2)
 
 What I mean is being able to have everything inside the top level collapsed as well.
 
 So that when I expand "1" and "2" the contents inside of it is also correctly folded
 
 like this

![image](https://user-images.githubusercontent.com/51902730/107812445-ae526480-6d3d-11eb-8ed8-5f6e3dc62774.png)

Thank you so much again, being able to use your plugin is one of my favorite aspect of Obsidian

**Comment by jawspeak** [comment]
I agree this would be awesome for navigating easier in a large mindmap.

This would be useful, and it seems that the underlying library `markmap` supports `initialExpandLevel`. https://markmap.js.org/docs/json-options#initialexpandlevel. Example https://markmap.js.org/repl, look at Expand all levels and Limit expand level.

Any pointers how you would want someone to contribute this @lynchjames if we sent in a PR? 

I also imagine how this could be a command palette type command to fold/unfold to level X. Similar to this issue in the visual code markmind mindmap plugin https://github.com/markmap/markmap-vscode/issues/19, so implementation may be similar too.

**Comment by jawspeak** [comment]
See also PR https://github.com/lynchjames/obsidian-mind-map/pull/89

**Comment by jawspeak** [comment]
I implemented this with the following 2 PRs:
- https://github.com/lynchjames/obsidian-mind-map/pull/102 (upgrade mindmap, to make the next possible)
- folding shortcuts / commands (open to suggestions how to implement the commands, I was inspired by how visual studio code does shortcuts) https://github.com/lynchjames/obsidian-mind-map/pull/103
